### PR TITLE
Update feature gates for 4.0 new design

### PR DESCRIPTION
**Pull request by mburke5678**
https://gist.github.com/rphillips/4545b9ad23c1ffcaef6b63f282177e18
https://github.com/deads2k/api/blob/044292b2cd938dfaf777bcab375d661fd798c373/config/v1/types_feature.go

### CODE REVIEW

File: modules/nodes-pods-disabling-features-cluster.adoc [hunk 1]
```diff
@@ -5,33 +5,42 @@
...
-  name: cluster-feature-gates
+  name: cluster
+  namespace: openshift-config
 spec:
-  enabled:           <1>
-  - DynamicAuditing
-  disabled:          <2>
-  - CustomPodDNS
+  featureSet: "TechPreviewNoUpgrade"
+     Enabled:  <1>
```
Comment (sjenning) [inline modules/nodes-pods-disabling-features-cluster.adoc]: everything on this line and later needs to be remove from the CR.  `featureSet` is a string field now that is either `Default` or `TechPreviewNoUpgrade`.  Those correspond to predetermined (by us) lists of a feature gates.

Comment (mburke5678) [reply modules/nodes-pods-disabling-features-cluster.adoc]: @sjenning Do you have an updated CR that I can use?

**Comment by mburke5678** [comment]
@rphillips  @sjenning  Please take a look.

Who is the QE resource who can review this PR?

**Comment by rphillips** [comment]
Looks great. lgtm. @sjenning?

**Comment by rphillips** [comment]
@mburke5678 after this PR goes in, the cluster resource will be installed by default: https://github.com/openshift/installer/pull/1546

**Comment by sjenning** [comment]
LGTM

**Comment by mburke5678** [comment]
@xingxingxia Hello. Is this a PR that you can review? I think Feature Gates is part of the Master team.

**Comment by xingxingxia** [comment]
@mburke5678 hello, @weinliu will be reviewing it.

**Comment by weinliu** [comment]
@mburke5678 , I'm done with the review work, the doc LGTM

**Comment by sunilcio** [comment]
Is there a plan to allow users to enable or disable individual feature?

**Comment by mburke5678** [comment]
@sunilcio As I understand, there will not be a way to enable individual features for 4.1. It will be all or none. @sjenning please let us know if i have this incorrect.

**Comment by rphillips** [comment]
That is correct. A customer will be able to enable "TechPreview' to enable extra features.

**Pull request merged**